### ISSUE TITLE
Critical vulnerability in `@xmldom/xmldom@0.8.3`

### ISSUE THREAD

**Issue by timhere** [kettanaito/msw#1447]
Hello, we have seen a warning via aquasec about a vulnerability in `xmldom`: https://avd.aquasec.com/nvd/2022/cve-2022-39353/

This is a dependency of `@mswjs/interceptors`, hopefully we can update `xmldom` from `0.8.3` to `0.8.4` to remove the issue.

See here: https://github.com/xmldom/xmldom/security/advisories/GHSA-crh6-fp67-6883

**Comment by timhere** [comment]
Upstream dependency fixed in https://github.com/mswjs/interceptors/pull/295 awaiting release

**Comment by kettanaito** [comment]
The upstream dependency has been released. Re-install `msw` and you will get the latest dependency version.

**Comment by Freddie-Pike** [comment]
While the upsream dependency has been fixed, if I reinstall the `xmldom` version is still not the newest. Is there a plan to do a release with this new security fix? 

If I'm confused here, lmk but this what my package-lock.json looks like!

```
"node_modules/@mswjs/interceptors": {
      "version": "0.17.5",
      "resolved": "https://registry.npmjs.org/@mswjs/interceptors/-/interceptors-0.17.5.tgz",
      "integrity": "sha512-/uZkyPUZMRExZs+DZQVnc+uoDwLfs1gFNvcRY5S3Gu78U+uhovaSEUW3tuyld1e7Oke5Qphfseb8v66V+H1zWQ==",
      "dev": true,
      "dependencies": {
        "@open-draft/until": "^1.0.3",
        "@types/debug": "^4.1.7",
        "@xmldom/xmldom": "^0.7.5",
        "debug": "^4.3.3",
        "headers-polyfill": "^3.1.0",
        "outvariant": "^1.2.1",
        "strict-event-emitter": "^0.2.4",
        "web-encoding": "^1.1.5"
      },
      "engines": {
        "node": ">=14"
      }
    },
```

Also, I'm using npm here btw, not yarn!

**Comment by timhere** [comment]
I don't think reinstalling `msw` will fix things because although [`@msjs/interceptors` ](https://github.com/mswjs/msw/blob/main/package.json#L90) has merged the PR to update `xmldom` to `0.8.5`, it hasn't triggered a release yet.

So this means for the moment you'll still get [`v0.17.6` of the interceptors dep](https://github.com/mswjs/interceptors/releases/tag/v0.17.6) which still has the vulnerability pinned in its lockfile.

If you need a temporary fix, you should be able to add this to your `package.json` resolutions field and reinstall with your package manager:

```
resolutions: {
  "msw/**/@xmldom/xmldom": "0.8.5"
}
```

Slightly confusing is that there was a previous vulnerability with `xmldom` that has been fixed: https://github.com/mswjs/msw/issues/1434

**Comment by kettanaito** [comment]
The interceptors don't have to be released at all for you to get the fix for the transient dependency of `xmldom`:

- msw
  - whichever patch version of `@mswjs/interceptors`
    - whichever patch version of `xmldom`

—[Source](https://github.com/mswjs/interceptors/blob/15eafa6215a328219999403e3ff110e71699b016/package.json#L71)

Installing `msw` gets you `xmldom@^0.8.3`, which means `0.8.x`. The fix is at `xmldom@0.8.5`, so you will get it per semver. Just make sure you're not hitting your package manager's cache when reinstalling (`rm -rf node_modules` or `--force`).

**Comment by ben-wilson-peak** [comment]
For anyone who stumbles on this while looking to patch:

```
  "resolutions": {
    "msw/**/@mswjs/interceptors": "0.19.0"
  }
```
